Core/Core/ViewModifiers/TestIdentifier.swift

```diff
@@ -0,0 +1,140 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+@available(iOSApplicationExtension 13.0, *)
+public struct TestTree {
+    public let kind: Kind?
+    public let id: String?
+    public let type: Any.Type
+    public let wrappedInfo: AnyEquatable?
+    public let subtrees: [TestTree]
+
+    public var info: Any? { wrappedInfo?.value }
+    public func info<E: Equatable>(_ key: String) -> E? {
+        (info as? [String: Any])?[key] as? E
+    }
+
+    public init<Info: Equatable>(kind: Kind?, id: String?, type: Any.Type, info: Info?, subtrees: [TestTree]) {
+        self.kind = kind
+        self.id = id
+        self.type = type
+        self.wrappedInfo = AnyEquatable(info)
+        self.subtrees = subtrees
+    }
+
+    public struct AnyEquatable {
+        public let value: Any
+        public let isEqual: (Any) -> Bool
+
+        public init<E: Equatable>(_ value: E) {
+            self.value = value
+            isEqual = { value == $0 as? E }
+        }
+        public init?<E: Equatable>(_ value: E?) {
+            guard let value = value else { return nil }
+            self.init(value)
+        }
+    }
+
+    /// Roughly equivalent to a CSS class
+    public enum Kind {
```

Comment:
The type of a view is often less than useful due to modifiers and generics. This allows for marking all of a class of elements. I just didn't want to call it "class" because that's a keyword and also easily confused with "type"